JASP-Engine/JASP/R/commonMachineLearningRegression.R

```diff
@@ -427,6 +427,12 @@
   return(x)
 }
 
+.scaleNumericData.default <- function(x, center = TRUE, scale = TRUE) {
+  warning(".scaleNumericData.default was called! you probably didn't want that!")
```

Comment:
Shouldn't this just fail hard? I could imagine you might get some unexpected values/bugs otherwise

Reply:
Well.. it failed hard before. `.scaleNumericData` has three methods that are the only ones that should ever be called:

- `.scaleNumericData.data.frame`, automatically handles the variable types in the data.frame.
- `.scaleNumericData.matrix`, assumes you put in numeric data and otherwise crashes.
- `.scaleNumericData.numeric`, obviously numeric.

It can happen that someone writes: `.scaleNumericData(myDataframe[, predictors])`. Now, if `predictors` has length 1 then R (annoyingly) simplifies the result and instead of a data.frame you pass a vector with some type to `.scaleNumericData`. If this vector contains numeric data, all is well, otherwise you get an error. 

As a fallback I now added `.scaleNumericData.default`, since all the cases where `.scaleNumericData` should actually do something are accounted for (I hope). That way, character/ factor data won't cause a crash. 

Another way to avoid this error is to enforce `drop = FALSE`. For example, `.scaleNumericData(myDataframe[, predictors, drop = FALSE])`. However, whenever we forget about this a bug appears. To be realistic, at some point we will forget about this.

So that gives two options:
1. be strict about `drop = FALSE` (good idea anyway I think).
2. have a fallback to avoid crashes when people didn't do `drop = FALSE`, possibly with a warning.

I don't really see how `.scaleNumericData.default` could introduce unexpected bugs, but maybe you have an idea? If you don't like the fallback idea we can also just remove it.

Reply:
Ah, I see. In that case I think you could actually remove the warning, but then you should also not have `.matrix` crash if it's character. Because then the behaviour is consistent: either you supply invalid data and it does nothing, or your data is (partially) valid and it's scaled